scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -192,6 +193,8 @@ sealed trait Execution[+T] extends java.io.Serializable {
    */
   def zip[U](that: Execution[U]): Execution[(T, U)] =
     Zipped(this, that)
+
+  override val hashCode = super.hashCode
```

Comment:
can't we just manually add a strict hashCode on all the subclasses that take `Executions`? That will stop it from recursing forever, right?

Of course, it does not address equality checks (but this PR didn't do that anyway).